src/permutation/constants.rs

```diff
@@ -1,20 +1,21 @@
-#![allow(non_snake_case)]
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+#![allow(non_snake_case)]
```

Comment:
Remove this from here and just add it in `lib.rs` to avoid repetition.

Reply:
Disabling lints in a precise way is not repetition, as each one should be used to indicate some important convention in that region of code. Naming lints can also be used as compile-time sanity checking.